community: added community visibility labels to header.

alejandromumo opened the pull request:
closes https://github.com/inveniosoftware/invenio-communities/issues/732

jennur commented:
Maybe this could go in another issue, but about the cramped space on mobile... we recently introduced `mr-0-mobile` and `ml-0-mobile` which could go on to the `.page-subheader`-grid. It would free up some space for the content (we have a lot of spacing to the edge of the screen on mobile, due to paddings and margins on nested containers).

Still, I agree with @jrcastro2 that maybe the label could go on top on mobile? 

On desktop, maybe we could just force the width of the label to make sure it doesn't break, and align it on top with the title, like this?:
<img width="1098" alt="Screenshot 2022-06-29 at 15 22 15" src="https://user-images.githubusercontent.com/21052053/176447038-fe6e4693-982f-46a1-9d08-acabc4417bcb.png">

What do you think?

jennur commented on invenio_communities/templates/semantic-ui/invenio_communities/details/header.html:
```diff
@@ -19,7 +19,17 @@
+          <div class="flex align-items-center rel-mb-1">
+            <h2 class="ui header mb-0">{{ community.metadata.title }}</h2>
+            {% if community.access.visibility == 'restricted' %}
+              <div class="rel-ml-1 align-self-baseline">
```
```suggestion
              <div class="rel-ml-1">
```
I think this is actually not needed if you remove the `align-items-center` from the `flex`-element? Also, I think the empty `<div>` around the `.flex`-element can be removed. Correct me if I'm wrong!

alejandromumo commented:
> If a community has a long name, the label doesn't look as nice, see image:
Maybe setting `width: max-content` to the label could fix it (it's just an idea, probably not the best)


When the title has large words, the whole page breaks in mobile. I opened this [issue](https://github.com/inveniosoftware/invenio-app-rdm/issues/1770)

Changes:

- Moved the label to be inline with the title, following @jennur suggestion. 
- Added `width: max-content` for `access-status` labels, tested it in other views (e.g. deposit form) and seems to be working fine. Nonetheless, if it might break something please let me know.
- Following @kpsherva suggestion, removed the access status label for `public`.  Nonetheless I think we have this behavior elsewhere (see: https://github.com/inveniosoftware/invenio-communities/blob/5ef4cb6048f39bd07d482d3eb787e923565c5b4c/invenio_communities/assets/semantic-ui/js/invenio_communities/details_search/components.js#L83-L88)
- Also added `mr-0-mobile` and `ml-0-mobile` for the subheader page as @jennur suggested.  

Result:

Mobile 

![Screenshot at Jun 30 10-43-46](https://user-images.githubusercontent.com/21204744/176633790-895defd1-9f98-486f-894c-d43d90964bc1.png)

Desktop

![Screenshot at Jun 30 10-52-34](https://user-images.githubusercontent.com/21204744/176635648-a959e19f-b196-4fa7-a220-897614510625.png)
